tests/thread_race/main.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     test
+ * @{
+ *
+ * @file
+ * @brief       Test for race conditions in context switching
+ *
+ * May have false negatives.
+ *
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * @}
+ */
+
+#include <avr/cpufunc.h>
+#include <avr/interrupt.h>
+#include <stdio.h>
+
+#include "irq.h"
+#include "sched.h"
+#include "thread.h"
+
+/* board.h for AVR_CONTEXT_SWAP_TRIGGER */
+#include "board.h"
+
+char iqr_check_stack[THREAD_STACKSIZE_DEFAULT];
+
+static volatile uint8_t irq_occurred;
+
+static void *_thread_irq_check(void *arg)
+{
+    (void)arg;
+
+    while (1) {
+        irq_occurred = 1;
+
+        thread_sleep();
+    }
+
+    return NULL;
+}
+
+/* Mostly copied from thread_wakeup() */
+static void _thread_wake_wo_yield(kernel_pid_t pid)
+{
+    unsigned old_state = irq_disable();
+
+    thread_t *other_thread = (thread_t *) thread_get(pid);
+
+    sched_set_status(other_thread, STATUS_RUNNING);
+
+    irq_restore(old_state);
+}
+
+static void _spin(void)
+{
+    /* Volatile so it is not messed with by optimizations */
+    volatile uint8_t i;
+
+    for (i=0; i<255; i++);
```

Comment:
Here and below (for the if-else statements), please follow coding conventions. You may also run uncrustify on this file to correct it automatically.

Reply:
Sorry, still new to RIOT conventions. Uncrustified.